src/spawner/spawner.py

```diff
@@ -1,9 +1,56 @@
 from abc import abstractmethod
 
+import marshmallow as marsh
+from peewee import ForeignKeyField
+
+from src.base_model import BaseModel
 from src.component import Component
 from src.schedule.schedule import Schedule
 
 
+class SpawnerConfiguration(BaseModel):
+    """Class representing a spawner configuration. It holds a list of
+    Schedules which is handled in the reference table class `SpawnerConfigurationXSchedule`.
+    This class has no fields except the `id` which is needed by the `Spawner`.
+    """
+
+    class Schema(BaseModel.Schema):
+        """Marshmallow schema for SpawnerConfiguration"""
+
+        def _make(self, data: dict) -> "SpawnerConfiguration":
+            """Constructs a SpawnerConfiguration from a dictionary.
+
+            :param data: The dictionary.
+            :return: A SpawnerConfiguration.
+            """
+            return SpawnerConfiguration(**data)
+
+
+class SpawnerConfigurationXSchedule(BaseModel):
+    """Reference table class for m:n relation
+    between SpawnerConfiguration and Schedule.
+    """
+
+    class Schema(BaseModel.Schema):
+        """Marshmallow schema for SpawnerConfigurationXSchedule"""
+
+        spawner_configuration_id = marsh.fields.UUID(required=True)
+        schedule_id = marsh.fields.UUID(required=True)
+
+        def _make(self, data: dict) -> "SpawnerConfigurationXSchedule":
+            """Constructs a SpawnerConfigurationXSchedule from a dictionary.
+
+            :param data: The dictionary.
+            :return: A SpawnerConfigurationXSchedule.
+            """
+            return SpawnerConfigurationXSchedule(**data)
+
+    spawner_configuration_id = ForeignKeyField(
+        SpawnerConfiguration, null=False, backref="schedules"
+    )
+    schedule_id = ForeignKeyField(Schedule, null=False)
```

Comment:
Does this mean that a spawner must have at least one schedule (and the reverse is also true)? If yes, does this work with the API (can you create a spawner config and a schedule simultaneously?)

Reply:
Interesting Question! 👀